src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/ExtractPrimerTask.cpp

```diff
@@ -0,0 +1,193 @@
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "ExtractPrimerTask.h"
+#include "PCRPrimerDesignForDNAAssemblyTask.h"
+
+#include <QApplication>
+#include <QDir>
+#include <QFile>
+
+#include <U2Core/AppContext.h>
+#include <U2Core/AppSettings.h>
+#include <U2Core/AnnotationTableObject.h>
+#include <U2Core/BaseDocumentFormats.h>
+#include <U2Core/DNASequenceObject.h>
+#include <U2Core/DocumentModel.h>
+#include <U2Core/DocumentUtils.h>
+#include <U2Core/GUrlUtils.h>
+#include <U2Core/IOAdapter.h>
+#include <U2Core/L10n.h>
+#include <U2Core/LoadDocumentTask.h>
+#include <U2Core/SaveDocumentTask.h>
+#include <U2Core/U2DbiRegistry.h>
+#include <U2Core/U2SafePoints.h>
+#include <U2Core/U2SequenceDbi.h>
+#include <U2Core/U2SequenceUtils.h>
+#include <U2Core/UserApplicationsSettings.h>
+
+namespace U2 {
+
+const QString ExtractPrimerTask::BACKBONE_ANNOTATION_NAME = "Backbone";
+const QString ExtractPrimerTask::RESULT_ANNOTATION_GROUP_NAME = "misc_feature";
+
+ ExtractPrimerTask::ExtractPrimerTask(const ExtractPrimerTaskSettings &_settings)
+    : Task(tr("Extract Primer Task"), TaskFlags_FOSE_COSC), settings(_settings) {
+}
+
+ExtractPrimerTask::~ExtractPrimerTask() {
+    delete result;
+}
+
+void ExtractPrimerTask::run() {
+    IOAdapterFactory *iof = AppContext::getIOAdapterRegistry()->getIOAdapterFactoryById(BaseIOAdapters::LOCAL_FILE);
+    SAFE_POINT_EXT(nullptr != iof, setError(L10N::nullPointerError("IOAdapterFactory")), );
+
+    DocumentFormat *format = AppContext::getDocumentFormatRegistry()->getFormatById(BaseDocumentFormats::PLAIN_GENBANK);
+    SAFE_POINT_EXT(nullptr != format, setError(L10N::nullPointerError("Genbank Format")), );
+    QVariantMap hints;
+    QScopedPointer<Document> doc(format->createNewLoadedDocument(iof, settings.outputFileUrl, stateInfo, hints));
+    CHECK_OP(stateInfo, );
+
+    U2DbiRef dbiRef = AppContext::getDbiRegistry()->getSessionTmpDbiRef(stateInfo);
+    CHECK_OP(stateInfo, );
+
+    DNASequence productSequence = getProductSequence();
+    CHECK_OP(stateInfo, );
+    U2EntityRef productRef = U2SequenceUtils::import(stateInfo, dbiRef, productSequence);
+    CHECK_OP(stateInfo, );
+
+    QList<SharedAnnotationData> annotations;
+    int resultIndex = PCRPrimerDesignForDNAAssemblyTask::FRAGMENT_INDEX_TO_NAME.indexOf(settings.fragmentName);
+    bool isForvard = resultIndex % 2 == 0;
+
+    U2Region fragmentRegion = U2Region(0, productSequence.length());
```

Comment:
The compiler isn't a dumb, it could handle, that there are an extra operation here and fix it on it's own. It won't call a copy operator in this case, the usual constructor only. But OK, I'll fix it